### PR TITLE
Add unit property for MetricData

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 1]
```diff
@@ -59,6 +61,29 @@
...
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.Volts.toString(), MetricUnit.V.toString());
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.Celsius.toString(), MetricUnit.C.toString());
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.Fahrenheit.toString(), MetricUnit.F.toString());
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.KW.toString(), MetricUnit.KW.toString());
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.Percent.toString(), MetricUnit.PERCENT.toString());
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.KBps.toString(), MetricUnit.KBps.toString());
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.Mhz.toString(), MetricUnit.Mhz.toString());
+      databaseUnitAndOutputUnitMap.put(RealtimeDataUnit.KB.toString(), MetricUnit.KB.toString());
+      databaseUnitAndOutputUnitMap.put(MetricUnit.A.toString(), MetricUnit.A.toString());
+      databaseUnitAndOutputUnitMap.put(MetricUnit.AMPS.toString(), MetricUnit.A.toString());
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:78]: double check if some place use amps, change to A

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 2]
```diff
@@ -59,6 +61,29 @@
    private static final int SERVER_ASSET_NAME_TIME_OUT = 7200;
    private static final int LIMIT_RESULT = 100;
    private static final String UnknownOulet = "unknownoutlet";
+   private static final int timestampArrayLength = 2;//The extraInfo of peakPower and minimumPower metric include since time and event time.
+   private static Map<String, String> databaseUnitAndOutputUnitMap = null;
+   static {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:66]: this map will be deprecated in v1.3

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 3]
```diff
@@ -222,18 +247,62 @@
...
          List<String> metricsNameList = assetIdAndMetricsNameList.get(realtimeDataEntry.getKey());
          for (RealTimeData realTimeData : realtimeDataEntry.getValue()) {
             for (ValueUnit valueUnit : realTimeData.getValues()) {
                if (metricsNameList.contains(valueUnit.getKey())) {
                   metricData = new MetricData();
+                  long timestamp = 0l;
+                  String metricName = valueUnit.getKey();
+                  String extraInfo = valueUnit.getExtraidentifier();
+                  if(extraInfo == null && specialMetricNames.contains(metricName)) {
+                     logger.error("The value of {} is invalid", metricName);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:266]: add id in the log.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 4]
```diff
@@ -222,18 +247,62 @@
...
+                     continue;
+                  }
+                  switch (metricName) {
+                  case MetricName.SERVER_AVERAGE_USED_POWER:
+                     //Time of AvgPower is since time
+                     timestamp = Long.valueOf(extraInfo);
+                     break;
+                  case MetricName.SERVER_PEAK_USED_POWER:
+                     String[] sinceTimeAndPeakTime = extraInfo.split(FlowgateConstant.SEPARATOR);
+                     if(sinceTimeAndPeakTime.length < timestampArrayLength) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:276]: upper case

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 1 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 2 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 4 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 5]
```diff
@@ -493,6 +570,11 @@ public static AssetIPMapping parseAssetIPMapingByString(String contentString) {
          data.setValueNum(value.getValueNum());
          data.setValue(value.getValue());
          data.setMetricName(value.getKey());
+         String unit = databaseUnitAndOutputUnitMap.get(value.getUnit());
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:573]: add comment that those kind of translation will be deprecated in next version.

File: flowgate-common/src/main/java/com/vmware/flowgate/common/model/ValueUnit.java [hunk 6]
```diff
@@ -62,12 +62,22 @@ public void setKey(String key) {
...
-       C("TEMPRETURE",1), F("TEMPRETURE",1), PERCENT("PERCENT",1), BTUPerHr("POWER",0.293071),
-       KB("STORAGE",1), Mhz("FREQUENCY", 1), KBps("RATE", 1);
+       //These units include MV,MW,MA,KW,KA,KWH,KB,KBps,PERCENT will be deprecated in Flowgate-1.3
+       MV("VOLTAGE",0.001), KV("VOLTAGE",1000),MW("POWER",0.001), KW("POWER",1000),
+       KWH("ENERGY",1000), MA("CURRRNT",0.001), KA("CURRENT",1000), PERCENT("PERCENT",1),
+       KB("STORAGE",1), KBps("RATE", 1),
+
+       A("CURRENT",1), V("VOLTAGE",1), W("POWER",1),C("TEMPRETURE",1), VOLTS("VOLTAGE",1),
+       F("TEMPRETURE",1), BTUPerHr("POWER",0.293071), Mhz("FREQUENCY", 1), mV("VOLTAGE",0.001),
+       kV("VOLTAGE",1000), mW("POWER",0.001), kA("CURRENT",1000), percent("PERCENT",1),
```
Comment (yixingjia) [inline flowgate-common/src/main/java/com/vmware/flowgate/common/model/ValueUnit.java:79]: mW MA , double check if we don't use it anywhere, then just remove it .

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 5 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:574]: done

File: flowgate-common/src/main/java/com/vmware/flowgate/common/model/ValueUnit.java (hunk 6 above)
Comment (Pengpengwanga) [reply flowgate-common/src/main/java/com/vmware/flowgate/common/model/ValueUnit.java]: done

**Review by yixingjia — APPROVED** [review]

**Pull request merged**